### PR TITLE
Fix linter errors

### PR DESCRIPTION
**Pull request by kostind**
### Please check if the PR fulfills these requirements
- [ ] The commit message follows our Submission guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? 
Fixed linter errors.

### What is the current behavior? 
(You can also link to an open issue here)

### What is the new behavior?
(Define and describe any new functionality. Clarify if this is a feature change)

### Does this PR introduce a breaking change?
(What changes might users need to make in their application due to this PR?)

### Any Other information:
I've reverted error-reason for ERC20DividendCheckpoint, EtherDividendCheckpoint, ModuleFactory, SecurityToken since it's produces "out of gas" error.
So we still have 16 error-reason linter errors.

### CODE REVIEW

File: contracts/ModuleRegistry.sol [hunk 1]
```diff
@@ -346,6 +351,8 @@ contract ModuleRegistry is IModuleRegistry, EternalStorage {
      */
     function pause() external whenNotPaused onlyOwner {
         set(Encoder.getKey("paused"), true);
+        /*solium-disable-next-line security/no-block-members*/
+        /*solium-disable-next-line security/no-block-members*/
```
Comment (maxsam4) [inline contracts/ModuleRegistry.sol]:
```suggestion
```
We only need to add it once.

File: contracts/SecurityTokenRegistry.sol [hunk 2]
```diff
@@ -164,9 +164,15 @@ contract SecurityTokenRegistry is ISecurityTokenRegistry, EternalStorage {
      * @param _polyToken is the address of the POLY ERC20 token
      * @param _owner is the owner of the STR
      */
-    function initialize(address _polymathRegistry, address _STFactory, uint256 _stLaunchFee, uint256 _tickerRegFee, address _polyToken, address _owner) payable external {
+    function initialize(
+        address _polymathRegistry, address _STFactory, uint256 _stLaunchFee, uint256 _tickerRegFee, address _polyToken, address _owner)
```
Comment (maxsam4) [inline contracts/SecurityTokenRegistry.sol]: If splitting function like this, all variables must go in separate lines
Look at https://solidity.readthedocs.io/en/v0.4.25/style-guide.html for the style guide

File: contracts/SecurityTokenRegistry.sol [hunk 3]
```diff
@@ -637,7 +657,7 @@ contract SecurityTokenRegistry is ISecurityTokenRegistry, EternalStorage {
     */
     function changeTickerRegistrationFee(uint256 _tickerRegFee) external onlyOwner {
         uint256 fee = getUint(TICKERREGFEE);
-        require(fee != _tickerRegFee);
+        require(fee != _tickerRegFee, "Fee isn't equal to the registration fee");
```
Comment (maxsam4) [inline contracts/SecurityTokenRegistry.sol]:
```suggestion
        require(fee != _tickerRegFee, "Fee not changed");
```

File: contracts/SecurityTokenRegistry.sol [hunk 4]
```diff
@@ -648,7 +668,7 @@ contract SecurityTokenRegistry is ISecurityTokenRegistry, EternalStorage {
     */
     function changeSecurityLaunchFee(uint256 _stLaunchFee) external onlyOwner {
         uint256 fee = getUint(STLAUNCHFEE);
-        require(fee != _stLaunchFee);
+        require(fee != _stLaunchFee, "Fee isn't equal to the launch fee");
```
Comment (maxsam4) [inline contracts/SecurityTokenRegistry.sol]:
```suggestion
        require(fee != _stLaunchFee, "Fee not changed");
```

File: contracts/libraries/VersionUtils.sol [hunk 5]
```diff
@@ -52,9 +52,9 @@ library VersionUtils {
      * @return bool
      */
     function compareLowerBound(uint8[] _version1, uint8[] _version2) internal pure returns(bool) {
-        require(_version1.length == _version2.length);
+        require(_version1.length == _version2.length, "Versions should match");
```
Comment (maxsam4) [inline contracts/libraries/VersionUtils.sol]:
```suggestion
        require(_version1.length == _version2.length, "Input length mismatch");
```

File: contracts/libraries/VersionUtils.sol [hunk 6]
```diff
@@ -83,9 +83,9 @@ library VersionUtils {
      * @return bool
      */
     function compareUpperBound(uint8[] _version1, uint8[] _version2) internal pure returns(bool) {
-        require(_version1.length == _version2.length);
+        require(_version1.length == _version2.length, "Versions should match");
```
Comment (maxsam4) [inline contracts/libraries/VersionUtils.sol]:
```suggestion
        require(_version1.length == _version2.length, "Input length mismatch");
```
For consistency, we write "what went wrong" in error statements.

File: contracts/mocks/MockRedemptionManager.sol [hunk 7]
```diff
@@ -35,10 +35,11 @@ contract MockRedemptionManager is TrackedRedemption {
      * @param _value The number of tokens to redeem
      */
     function redeemTokenByOwner(uint256 _value) public {
-        require(tokenToRedeem[msg.sender] >= _value);
+        require(tokenToRedeem[msg.sender] >= _value, "Value is greater than available");
```
Comment (maxsam4) [inline contracts/mocks/MockRedemptionManager.sol]:
```suggestion
        require(tokenToRedeem[msg.sender] >= _value, "Insufficient tokens redeemable");
```

The error statements are printed when the require statement fails and reverts. In this case, the statement will be printed when `tokenToRedeem[msg.sender]` is LESS than `_value`. So the error statement must mention what exactly went wrong.

**Review by maxsam4 — CHANGES_REQUESTED** [review]
@kostind In a lot of require error messages, you have mentioned "what should happen" for the require statement to pass. For consistency, we should pass "what went wrong" in the error statement. I have added individual comments for starting few files but there are a lot more of these. 

Also, the way you have broken down functions is against the style guide https://solidity.readthedocs.io/en/v0.4.25/style-guide.html

Linters are supposed to impose that style guide but linters are not perfect. Please read the style and refactor the function definitions that you have changed accordingly.
I will review this again after the changes.

File: contracts/SecurityTokenRegistry.sol [hunk 8]
```diff
@@ -538,7 +552,10 @@ contract SecurityTokenRegistry is ISecurityTokenRegistry, EternalStorage {
      * @param _tokenDetails is the off-chain details of the token
      * @param _deployedAt is the timestamp at which the security token is deployed
      */
-    function modifySecurityToken(string _name, string _ticker, address _owner, address _securityToken, string _tokenDetails, uint256 _deployedAt) external onlyOwner {
+    function modifySecurityToken(
+        string _name, string _ticker, address _owner, address _securityToken, string _tokenDetails, uint256 _deployedAt)
```
Comment (satyamakgec) [inline contracts/SecurityTokenRegistry.sol]: here also need to follow the style guide

File: contracts/interfaces/ISTFactory.sol [hunk 9]
```diff
@@ -16,6 +16,7 @@ interface ISTFactory {
      * @param _divisible whether the token is divisible or not
      * @param _polymathRegistry is the address of the Polymath Registry contract
      */
-    function deployToken(string _name, string _symbol, uint8 _decimals, string _tokenDetails, address _issuer, bool _divisible, address _polymathRegistry)
-        external returns (address);
+    function deployToken(
+        string _name, string _symbol, uint8 _decimals, string _tokenDetails,
```
Comment (satyamakgec) [inline contracts/interfaces/ISTFactory.sol]: Styling problem

File: contracts/interfaces/ISecurityTokenRegistry.sol [hunk 10]
```diff
@@ -23,7 +23,9 @@ interface ISecurityTokenRegistry {
      * @param _tokenDetails off-chain details of the token
      * @param _deployedAt Timestamp at which security token comes deployed on the ethereum blockchain
      */
-    function modifySecurityToken(string _name, string _ticker, address _owner, address _securityToken, string _tokenDetails, uint256 _deployedAt) external;
+    function modifySecurityToken(
+        string _name, string _ticker, address _owner,
```
Comment (satyamakgec) [inline contracts/interfaces/ISecurityTokenRegistry.sol]: Styling problem

File: contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol [hunk 11]
```diff
@@ -12,8 +12,11 @@ contract ERC20DividendCheckpoint is DividendCheckpoint {
 
     // Mapping to token address for each dividend
     mapping (uint256 => address) public dividendTokens;
-
-    event ERC20DividendDeposited(address indexed _depositor, uint256 _checkpointId, uint256 _created, uint256 _maturity, uint256 _expiry, address indexed _token, uint256 _amount, uint256 _totalSupply, uint256 _dividendIndex, bytes32 indexed _name);
+    event ERC20DividendDeposited(
+        address indexed _depositor, uint256 _checkpointId, uint256 _created, uint256 _maturity,
```
Comment (satyamakgec) [inline contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol]: Styling problem

**Review by satyamakgec — CHANGES_REQUESTED** [review]
The problem I see is the styling as @maxsam4 already mention I point out the same 2 - 3 places but It is present many other please as well

**Comment by kostind** [comment]
@maxsam4 We have require() without error messages in 4 contracts: ERC20DividendCheckpoint, EtherDividendCheckpoint, ModuleFactory, SecurityToken. I've added message, but reverted them since they produce "out of gas" error. I'm not sure how to fix them.

**Comment by maxsam4** [comment]
@kostind you can leave them as it is for now. 

We'll have to optimize them to reduce size. We can do that later, just skip those contracts for now. If you have time left after rest of changes, feel free to look for any possible optimizations :)

**Pull request merged**